demos/horizontal_text_detection/python/rtsp_client.py

```diff
@@ -0,0 +1,136 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cv2
+import time
+import argparse
+from functools import partial
+import tritonclient.grpc as grpcclient
+import numpy as np
+import subprocess #nosec
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--grpc_address', required=False, default='localhost:9000', help='Specify url to grpc service')
+parser.add_argument('--input_stream', required=False, default="rtsp://localhost:8080/channel1", type=str, help='Url of input rtsp stream')
+parser.add_argument('--output_stream', required=False, default="rtsp://localhost:8080/channel2", type=str, help='Url of output rtsp stream')
+parser.add_argument('--model_name', required=False, default="detect_text_images", type=str, help='Name of the model')
+parser.add_argument('--width', required=False, default=704, type=int, help='Width of model\'s input image')
+parser.add_argument('--height', required=False, default=704, type=int, help='Height of model\'s input image')
+parser.add_argument('--verbose', required=False, default=False, type=bool, help='Height of model\'s input image')
+parser.add_argument('--input_name', required=False, default="image", type=str, help='Name of the model\'s input')
+args = parser.parse_args()
+output_stream = args.output_stream
+
+triton_client = grpcclient.InferenceServerClient( url=args.grpc_address, verbose=False)
+
+WIDTH = args.width
+HEIGHT = args.height
+cap = cv2.VideoCapture(args.input_stream, cv2.CAP_FFMPEG)
+fps = cap.get(cv2.CAP_PROP_FPS)
+force_exit = False
+
+def decode(text):
+    word = ''
+    last_character = None
+    for character in text:
+        if character == last_character:
+            continue
+        elif character == '#':
+            last_character = None
+        else:
+            last_character = character
+            word += character
+    return word
+
+def get_text(output):
+    alphabet = '#1234567890abcdefghijklmnopqrstuvwxyz'
+    preds = output.argmax(2)
+    word = ''
+    for pred in preds:
+        word += alphabet[pred[0]]
+    return decode(word)
+
+def finish():
+    global force_exit
+    force_exit = True
+
+def grab_frame(cap):
+    success, frame = cap.read()
+    if not success:
+        print("[WARNING] No Input frame")
+        finish()
+        return None
+
+    if frame.shape[0] > HEIGHT and frame.shape[1] > WIDTH:
+        frame = cv2.resize(frame, (HEIGHT, WIDTH), interpolation=cv2.INTER_LINEAR)
+    else:
+        frame = cv2.resize(frame, (HEIGHT, WIDTH), interpolation=cv2.INTER_AREA)
```

Comment:
I think it should be reversed - INTER_LINEAR when we scale the frame up (it's smaller than HEIGHT and WIDTH) and INTER_AREA when we scale down.